app/src/main/java/woowacourse/shopping/ui/MainViewModel.kt

```diff
@@ -11,16 +13,16 @@ class MainViewModel(
     private val productRepository: ProductRepository,
     private val cartRepository: CartRepository,
 ) : ViewModel() {
```

Comment:
아마 이렇게 하면 될거에요

```kotlin
class MainViewModel(
    private val cartRepository: CartRepository
) : ScopeViewModel() {
    private val productRepository: ProductRepository by inject()

val viewModelScopeRepositoryModule = module {
    scope<MainViewModel> {
        scoped<ProductRepository> { InMemoryProductRepository() }
    }
}
```

[참고 링크 ](https://insert-koin.io/docs/reference/koin-android/scope/#viewmodel-scope-since-354)
![image](https://github.com/user-attachments/assets/726175d3-67f8-493a-b885-22f9cce50849)

Reply:
더 나은 방법이 있는지는 꼬상, 오둥이가 알려줄 듯..?